index.ts

```diff
@@ -5,41 +5,64 @@ import fetchRetry from 'fetch-retry';
 import isString from 'lodash.isstring';
 import md5 from 'md5';
 import ms from 'ms';
-import fetch from 'node-fetch';
+import fetch, { Response } from 'node-fetch';
 import removeTrailingSlash from 'remove-trailing-slash';
 import { v4 as uuid } from 'uuid';
 
 const version = require('./package.json').version;
 
 const retryableFetch = fetchRetry(fetch as any) as unknown as typeof fetch;
 const setImmediate = global.setImmediate || process.nextTick.bind(process);
-const noop = () => {};
 
-enum AnalyticsState {
+export type AnalyticsMessage = AnalyticsIdentity & {
+  context?: { [key: string]: unknown };
+  traits?: { [key: string]: unknown };
```

Comment:
`traits` is a [contextual field](https://docs.rudderstack.com/rudderstack-api/api-specification/rudderstack-spec/common-fields#contextual-fields) and should not be included in this type. it only exists on [group](https://docs.rudderstack.com/rudderstack-api/api-specification/rudderstack-spec/group) and [identify](https://docs.rudderstack.com/rudderstack-api/api-specification/rudderstack-spec/identify)